Week-3/Assignment 1/script.js

```diff
@@ -0,0 +1,37 @@
+function memoize(fn) {
+  const cache = new Map();
+  return function (...args) {
+    const key = args.toString();
+    if (cache.has(key)) {
+      return cache.get(key);
+    }
+    cache.set(key, fn(...args));
+    return cache.get(key);
+  };
+}
+
```

Comment:
The below add function can only accepts 2 variables hence we can only add 2 numbers. But as per Guideline 4 -> The "function should be able to take n number of argument" .